crates/cxx-qt-gen/src/generator/rust/constructor.rs

```diff
@@ -341,7 +341,8 @@ pub fn generate(
         );
```

Comment:
The issue here is in the second patch for CXX-Qt Constructor.

In the case of qml_features we're generating two methods, `route_arguments` and `new_rs`.
with these signatures (try `cargo expand properties` in the `examples/qml_features/rust` folder):
```
            unsafe extern "C" fn __route_arguments_rust_properties_1<'a>(
                arg0: bool,
                arg1: &'a QUrl,
                arg2: &'a QUrl,
                arg3: &'a QString,
            ) -> CxxQtConstructorArgumentsRustProperties1<'a> {
```
and:
```rust
            #[doc(hidden)]
            #[export_name = "cxx_qt_rust_properties$cxxbridge1$new_rs_rust_properties_1"]
            unsafe extern "C" fn __new_rs_rust_properties_1<'a>(
                args: CxxQtConstructorNewArgumentsRustProperties1<'a>,
            ) -> *mut RustPropertiesRust {
```
In both cases, the lifetimes could be elided (e.g. by just not adding them in the `route_arguments`, and by using `'_` in the second case).

This can be fixed by either changing our generation in that way, or just by adding `#[allow(clippy::needless_lifetimes)]`.
As the second is much easier, I'd just go with that. The generated code is still correct, it's just producing unneeded lifetimes.

Reply:
@mattkdab I've added a suggestion that should fix it :)

Reply:
In any case, quite interesting that Clippy didn't report this previously :thinking: 
I guess it does that because before it figured this was only caused by the macro, so it shouldn't be reported.
But with the span we're telling clippy: This was generated because of the `constructor.imp` span. Which coincidentally contains the same lifetime, with the same span.

It's a bit of a bummer, as we kinda want clippy to know that this is generated code, which it doesn't have to care about. But then again, we want to show that this specific piece of code was generated because of the added constructor impl...

For now just disabling the clippy warnings works, but it's a bit of a larger issue that may warant figuring out...